helper.py

```diff
@@ -318,19 +318,48 @@ def thunderbird_url(page, channel='None'):
 
 
 @jinja2.contextfunction
-def donate_url(ctx, content='', source='thunderbird.net', medium='give', campaign='donation_flow_2023', download=False):
-    # If this link is from a download button, donate.mozilla.org has thank you text.
-    download_string = ''
-    if download:
-        download_string = '&tbdownload=true'
-
-    # Add utm_campaign if we are using it.
-    campaign_string = ''
-    if campaign:
-        campaign_string = '&utm_campaign={0}'.format(campaign)
-
-    return (settings.DONATE_LINK.format(content=content, source=source, medium=medium)
-            + campaign_string + download_string)
+def donate_url(ctx, content='', source='thunderbird.net', medium='fru', campaign='donation_2023', show_donation_modal=True, download=None, download_channel=None):
+    """Forms a donation url with the given parameters. If you pass in None for any of the fields they will be excluded from the url
+    :param content: UTM Content tag
+    :param source: UTM Source tag
+    :param medium: UTM Medium tag
+    :param campaign: UTM Campaign tag
+    :param show_donation_modal: Whether we want to append form=support that will automatically load the FRU modal
+    :param download: Whether we have already downloaded Thunderbird (Download button specific.) Boolean or None.
+    :param download_channel: What download channel to append to the url (Download button specific.) String or None.
+    """
+    form = None
+    if show_donation_modal:
+        form = 'support'
+
+    query = {
+        'utm_content': content,
+        'utm_source': source,
+        'utm_medium': medium,
+        'utm_campaign': campaign,
+        'form': form,
+        'downloaded': download,
+        'download_channel': download_channel
+    }
+
+    query_list = []
+
+    for item in query.items():
```

Comment:
dictionary -> url parameters is a common operation and can be done with `urllib.urlencode` instead. `urllib.parse.urlencode` in Python 3.

Reply:
Thanks for the catch! Now if only there was a dict filter for None values 😅

Reply:
Yeah, you need to remove the `None`s, but a dict comprehension is a good way to do this:
`filtered_query = {k: v for k, v in query.items() if v is not None}`